x/gamm/keeper/pool_test.go

```diff
@@ -316,3 +335,96 @@ func (suite *KeeperTestSuite) TestGetPoolAndPoke() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestSetStableSwapScalingFactors() {
+	suite.SetupTest()
+
+	pk := ed25519.GenPrivKey().PubKey()
+	addr := sdk.AccAddress(pk.Address())
+
+	failPk := ed25519.GenPrivKey().PubKey()
+	failAddr := sdk.AccAddress(failPk.Address())
+
+	defaultPoolId := uint64(1)
+
+	testcases := []struct {
+		name             string
+		poolId           uint64
+		scalingFactors   []uint64
+		sender           sdk.AccAddress
+		expError         error
+		isStableSwapPool bool
+	}{
+		{
+			name:             "Error: Pool does not exist",
+			poolId:           defaultPoolId + 1,
+			scalingFactors:   []uint64{1, 1},
+			sender:           addr,
+			expError:         types.PoolDoesNotExistError{PoolId: defaultPoolId + 1},
+			isStableSwapPool: false,
+		},
+		{
+			name:             "Error: Pool id is not of type stableswap pool",
+			poolId:           defaultPoolId + 1,
+			scalingFactors:   []uint64{1, 1},
+			sender:           addr,
+			expError:         fmt.Errorf("pool id 2 is not of type stableswap pool"),
+			isStableSwapPool: false,
+		},
+		{
+			name:             "Error: can not set scaling factors",
+			poolId:           defaultPoolId + 2,
+			scalingFactors:   []uint64{1, 1},
+			sender:           failAddr,
+			expError:         types.ErrNotScalingFactorGovernor,
+			isStableSwapPool: true,
+		},
+		{
+			name:             "Valid case",
+			poolId:           defaultPoolId + 3,
+			scalingFactors:   []uint64{1, 1},
+			sender:           addr,
+			isStableSwapPool: true,
+		},
+	}
+	for _, tc := range testcases {
+		suite.Run(tc.name, func() {
+			// Mint some assets to the accounts.
+			if tc.isStableSwapPool == true {
+				suite.FundAcc(tc.sender, defaultAcctFundsStableSwap)
+				poolId, err := suite.App.GAMMKeeper.CreatePool(
+					suite.Ctx,
+					stableswap.NewMsgCreateStableswapPool(
+						tc.sender,
+						defaultPoolParamsStableSwap,
+						defaultPoolAssetsStableSwap,
+						tc.scalingFactors,
+						defaultFutureGovernor))
+
+				suite.Require().NoError(err)
+				pool, err := suite.App.GAMMKeeper.GetPoolAndPoke(suite.Ctx, poolId)
+				stableswapPool, _ := pool.(*stableswap.Pool)
+				stableswapPool.ScalingFactorController = tc.sender.String()
```

Comment:
Hey @p0mvn, I have tried to pass the valid case but somehow I can't set  `ScalingFactorController ` for the Pool. As you can see in this line, I have set the hardcode for `Pool` but when I check `ScalingFactorController` again of this `Pool` at this line it is null. Can you check this and tell me if I am wrong at any steps   https://github.com/osmosis-labs/osmosis/blob/30460bacef932b03854f59938aa1d29db9a8f99b/x/gamm/keeper/pool.go#L277,